extensions/amp-story/1.0/amp-story.js

```diff
@@ -2467,6 +2470,30 @@ export class AmpStory extends AMP.BaseElement {
       extensionsFor.installExtensionForDoc(ampdoc, 'amp-analytics');
     }
   }
+
+  /**
+   * If there are inline localization strings, register as current document language.
+   */
+  registerInlineLocalizationStrings_() {
+    const inlineStringsEl = this.win.document.querySelector(
+      'script[amp-localization="amp-story"]'
+    );
+    if (
+      !inlineStringsEl ||
+      inlineStringsEl.getAttribute('i-amphtml-version') !=
+        getMode(this.win).rtvVersion
+    ) {
+      return;
+    }
+    const stringsOrNull = tryParseJson(inlineStringsEl.textContent);
+    if (!stringsOrNull) {
+      return;
+    }
+    getLocalizationService(this.element).registerLocalizedStringBundle(
+      this.win.document.querySelector('[lang]')?.getAttribute('lang') || 'en',
```

Comment:
Please make sure the `lang` always comes from a single source, ideally in the localization service.

You're missing some logic here, like the viewer language overrides

https://github.com/ampproject/amphtml/blob/bc4e8cf176603cc4f6786ef9f14d0ff41d7c81d0/src/service/localization/index.js#L103-L105

Reply:
I can use `getLanguageCodesForElement(this.element)[0]`